opm/simulators/flow/rescoup/ReservoirCouplingSlave.hpp

```diff
@@ -73,7 +88,22 @@ class ReservoirCouplingSlave {
     bool activated_{false};
     std::string slave_name_;  // This is the slave name as defined in the master process
     ReservoirCoupling::Logger logger_;
+    // Order of the slave groups. A mapping from slave group index to slave group name.
+    // The indices are determined by the order the master process sends us the group names, see
+    // receiveMasterGroupNamesFromMasterProcess_()
+    // Later, the master process will send us group name indices, and not the group names themselves,
+    // so we use this mapping to recover the slave group names from the indices.
+    std::map<std::size_t, std::string> slave_group_order_;
+    // Stores data that changes for a single report step or for timesteps within a report step.
+    std::unique_ptr<ReservoirCouplingSlaveReportStep<Scalar>> report_step_data_{nullptr};
 };
 
 } // namespace Opm
+
+// Include the complete definition after the class declaration to avoid circular dependency
+// while still providing complete types for unique_ptr destructor instantiation.
+// This is included here (after namespace closing) so that when the destructor is
+// instantiated during template instantiation, the complete type is available.
+#include <opm/simulators/flow/rescoup/ReservoirCouplingSlaveReportStep.hpp>
+
```

Comment:
Usually this should be resolved in another way (splitting into more headers).

What error were you getting?